test/jdk/javax/swing/JMenuBar/TestMenuMnemonicLinuxAndMac.java

```diff
@@ -117,5 +116,7 @@ private static void createAndShowUI() {
         frame.setLocationRelativeTo(null);
         frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
         frame.setVisible(true);
+        frame.toFront();
+        frame.requestFocus();
```

Comment:
If you are particular about focus being gained before the test starts, maybe you can use `CountDownLatch `and have FocusListener for frame and  wait till timeout and  release the latch in `focusGained`

Reply:
`CountDownlatch` should be used along with` requestFocus` ?
Otherwise, I think the test will timeout anyway because if we remove these two statements, frame may not have focus.

Reply:
Yes as the requestFocus spec says "Developers must never assume that this Component is the focus owner until this Component receives a FOCUS_GAINED event"
but I guess the present change is good enough for now since there is robot delay and all..